src/content/tools/hooks.md

```diff
@@ -0,0 +1,277 @@
+for their input and output. With hooks, the Dart SDK can:
+
+1. Discover your hooks.
+1. Execute your hooks with the necessary input.
+1. Consume the output produced by the hooks.
+
+Example project:
+
+```plaintext
+📁  example_project                   // project with hooks
+    📁  hook                          // add hook scripts here
+        📄 build.dart
+    📁  lib                           // reference your assets here
+        📄 example.dart
+    📁  src                           // add native sources here
+        📄 example_native_library.c
+        📄 example_native_library.h
+    📁  test                          // test your assets here
+        📄 example_test.dart
+```
+
+## Hooks
+
+Currently, the build hook is available, but more are
+planned. To learn more, see the following.
+
+### Build hooks {:.no_toc}
+
+A build hook builds assets that you want
+to run during your project's build process. This is useful
+for tasks that need to happen before your main
+application code is compiled, such as:
+
+* Compiling C, C++, Rust, or other native code into a
+  dynamic library.
+* Downloading or generating other assets required at
+  runtime.
+
+A package's build hook is placed in the `hook/build.dart`
+directory. Use the [`build`][] function to parse the hook
+input with [`BuildInput`] and write the hook output with
+[`BuildOutputBuilder`]. The hook should place downloaded and
+generated assets in [`BuildInput.sharedOutputDirectory`][].
+
+The build hooks are run in parallel with Dart compilation and may do longer
+running operations such as downloading or calling a native compiler. The assets
+produced in a build hook of a package may depend on [`assets`][] or
+[`metadata`][] produced by build hooks of direct dependencies of the package.
+Therefore, build hooks are run in the order of dependencies.
+
+[`assets`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput/assets.html
+[`build`]: {{site.pub-api}}/hooks/latest/hooks/build.html
+[`BuildInput`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput-class.html
+[`BuildOutputBuilder`]: {{site.pub-api}}/hooks/latest/hooks/BuildOutputBuilder-class.html
+[`BuildInput.sharedOutputDirectory`]: {{site.pub-api}}/hooks/latest/hooks/HookInput/outputDirectoryShared.html
+[`metadata`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput/metadata.html
+
+## Assets
+
+Assets are the files output from a hook that need to be
+bundled in a Dart application. Assets can be accessed
+at runtime from the Dart code. Currently, the Dart SDK can
+use the `CodeAsset` type, but more are planned.
+To learn more, see the following.
+
+### CodeAsset type {:.no_toc}
+
+A [`CodeAsset`][] represents a code asset. A code asset is
+a dynamic library compiled from a language other than
+Dart, such as C, C++, Rust, or Go. `CodeAsset` is part of
+the `code_asset` package and are called with `dart:ffi` from
+the Dart code at runtime using the `@Native` annotation.
+A code asset can be used with build and link hooks.
```

Comment:
Should we remove the referene to link hook here since we explain nowhere what that is yet?